v2/dtos/event.go

```diff
@@ -18,11 +18,11 @@ type Event struct {
 	common.Versionable `json:",inline"`
 	ID                 string        `json:"id"`
 	Pushed             int64         `json:"pushed,omitempty"`
-	DeviceName         string        `json:"deviceName"`
+	DeviceName         string        `json:"deviceName" validate:"required"`
 	Created            int64         `json:"created"`
 	Modified           int64         `json:"modified,omitempty"`
 	Origin             int64         `json:"origin"`
```

Comment:
I think  `Origin` be required?

Reply:
FYI, @tonyespy and I will bring this up in the Device Services WG meeting on Monday. I have verified that V1 doesn't require `Origin` is set, but IMHO that is a miss in V1, since if the device doesn't set it the Device Service does.

Reply:
In the interest in wrapping up `feat: Add POST event functions based for v2 API` edgexfoundry/edgex-go#2610. Leave `Origin` as is and we can deal with in a follow-up issue/pr.

Reply:
Why does `Modified` have "omitempty" but `Created` does not?

Reply:
`Modified `have "omitempty" because it is empty before updating
`Created `will always contain value once it is created
However, I thought we can remove `Modified` from `Event `and `Reading`.  We had it because it was in `DescribableObject`.
`Event `and `Reading `cannot be updated, and `Pushed` is enough for `Event`

Reply:
Remove `Modified` field from `Event` and `Reading` DTOs/models as @cloudxxx8 suggested.

Updated `Origin` as required and also applied in PR https://github.com/edgexfoundry/edgex-go/pull/2653.